addons/stock/wizard/product_replenish.py

```diff
@@ -85,7 +85,8 @@ def default_get(self, fields):
         if 'route_id' in fields and 'route_id' not in res and product_tmpl_id:
             res['route_id'] = self.env['stock.route'].search(self._get_route_domain(product_tmpl_id), limit=1).id
             if not res['route_id']:
-                res['route_id'] = product_tmpl_id.route_ids.filtered(lambda r: r.company_id == self.env.company or not r.company_id)[0].id
+                if product_tmpl_id.route_ids:
+                    res['route_id'] = product_tmpl_id.route_ids.filtered(lambda r: r.company_id == self.env.company or not r.company_id)[0].id
```

Comment:
The fix is incorrect, you could still do `[0]` on an empty recordset
```suggestion
                route = product_tmpl_id.route_ids.filtered(lambda r: r.company_id == self.env.company or not r.company_id)[0]
                res['route_id'] = route and route.id
```